### DISCUSSION TITLE
Example eslint configuration for Apollo frontend project

### DISCUSSION THREAD

**Discussion by MichaelKim0407** [graphql-hive/graphql-eslint#1849] [Q&A]
Hi, I have tried to read through the documentation but admittedly I don't have in-depth knowledge of ESLint or GraphQL, and I just need an example to make my project work.

My project: I have a `schema.graphql` generated by another project that I do not want to touch. My frontend is React/JavaScript + Apollo client. I have a bunch of `gql` queries and mutations in my `js(x)` files that I would like to validate against the schema file. Currently I'm relying on my IDE but I want to add the check to CI.

It seems that the examples provided in the [docs](https://the-guild.dev/graphql/eslint/docs/getting-started) are desigend to lint `.graphql` files, not `gql` code in `js(x)`. I tried to mess around but either I get no error being reported at all or the plugin complains about every javascript file as if they are graphql files.

An example or a direction would be appreciated. Thanks!

**Comment by eddiesholl** [comment]
I'm in a similar situation, I think I've go the exact same question. I'm in a next.js typescript app, we describe fragments in page.tsx files, and define queries in seperate gql.ts files. I can't find a config that allows the parser to execute across those files for running graphql eslint rules.
